src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -160,6 +160,8 @@ def extract_return_type_from_udtf_type_hints(
                     for name, column_type in zip(output_schema, column_type_hints)
                 ]
             )
+        else:  # both type hints and return type are specified
```

Comment:
returning `None` since type hints are ignored when a `StructType` is specified as return type.